### PR TITLE
Add blog post for security pipeline tools

### PR DESCRIPTION
**Pull request by bitkeks**

### CODE REVIEW

File: _i18n/en/_posts/blog/2024-08-21-security-tools-tech-preview.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  - "bitkeks"
+about:
+  - "90n20"
+  - "bitkeks"
+---
+
+## Motivation
+
+In today's rapidly evolving cloud landscape, security is paramount. As organizations increasingly rely on cloud infrastructure, the need for robust security measures is more critical than ever. This is particularly true for users of the Sovereign Cloud Stack (SCS), where regulatory requirements mandate regular penetration tests and security audits. Frameworks such as CIS Controls (formerly CIS18) and the NIS2 Directive impose stringent security standards, requiring organizations to proactively manage and mitigate cybersecurity risks.
+
```
Comment (bitkeks) [inline _i18n/en/_posts/blog/2024-08-21-security-tools-tech-preview.md:19]: Add a short sentence about the tech preview here:

```suggestion

To enable SCS users to run their own security tests inside their infrastructure, the SCS project conceptualized and implemented tooling for automated penetration testing. With release R7 this component is published as a tech preview and available for testing and experimenting.

```

File: _i18n/en/_posts/blog/2024-08-21-security-tools-tech-preview.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+Trivy Operator’s capability to export detailed security reports is a key feature, enabling teams to have a clear understanding of the security posture of their Kubernetes clusters. Additionally, DefectDojo serves as a central platform for managing vulnerabilities across the entire stack, providing a unified view that aids in prioritizing and remediating security issues. This centralization aligns with the best practices recommended by frameworks like ISO/IEC 27001 and the NIST Cybersecurity Framework (CSF).
+
+
+## Why security teams benefit from automated pentesting
+
+Understanding the difference between Static Application Security Testing (SAST) and Dynamic Application Security Testing (DAST) is crucial for security teams. SAST focuses on analyzing source code or binaries for vulnerabilities, while DAST involves testing running applications for security issues. The automated pentesting tools in SCS cater to both these methodologies, ensuring that security teams have comprehensive coverage.
```
Comment (bitkeks) [inline _i18n/en/_posts/blog/2024-08-21-security-tools-tech-preview.md:32]: Does the pipeline cover SAST? Where?

File: _i18n/en/_posts/blog/2024-08-21-security-tools-tech-preview.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+One of the challenges faced during implementation was the long runtime of certain jobs, particularly those involving Greenbone Vulnerability Management. Efforts have been made to optimize these jobs to reduce their impact on overall pipeline performance. The export functionality of Trivy Operator reports has also been streamlined, enabling easier integration with other security tools and processes, thereby ensuring ongoing compliance with regulatory requirements like CIS Controls and the NIST CSF.
+
+
+## Usage
+
+To utilize the automated pentesting tools in your own pipeline, you can follow the documentation provided in the Quickstart guide. This guide offers detailed instructions on how to integrate the tools into your CI/CD pipeline, ensuring that your infrastructure and applications are continuously tested for vulnerabilities, in line with compliance standards such as CIS Controls and NIS2.
```
Comment (bitkeks) [inline _i18n/en/_posts/blog/2024-08-21-security-tools-tech-preview.md:46]: Links to quickstart guides and repo :+1:

**Review by bitkeks — COMMENTED** [review]
LGTM, added some improvement suggestions

**Comment by bitkeks** [comment]
Current status: David works on an update locally and will push new commit in near time

File: _i18n/en/_posts/blog/2024-08-21-security-tools-tech-preview.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  - "90n20"
+  - "bitkeks"
+about:
+  - "90n20"
+  - "bitkeks"
+---
+
+## Motivation
+
+In today's rapidly evolving cloud landscape, security is paramount. As organizations increasingly rely on cloud infrastructure, the need for robust security measures is more critical than ever. This is particularly true for users of the Sovereign Cloud Stack (SCS), where regulatory requirements mandate regular penetration tests and security audits. Frameworks such as CIS Controls (formerly CIS18) and the NIS2 Directive impose stringent security standards, requiring organizations to proactively manage and mitigate cybersecurity risks.
```
[the remark this answers is not in the document]
Comment (90n20) [reply _i18n/en/_posts/blog/2024-08-21-security-tools-tech-preview.md]: Specify concertns about sensitive customers

File: _i18n/en/_posts/blog/2024-08-21-security-tools-tech-preview.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+The ability to consistently conduct thorough pentests ensures that the SCS platform remains compliant with these regulations, thereby reducing the risk of breaches and ensuring the integrity of the system. Furthermore, offering detailed security reports to end-customers can significantly enhance transparency, providing them with confidence in the platform's security posture. This approach not only fulfills compliance obligations but also establishes trust, which is crucial in the competitive cloud service provider (CSP) market.
```
[the remark this answers is not in the document]
Comment (90n20) [reply _i18n/en/_posts/blog/2024-08-21-security-tools-tech-preview.md]: Include the importance about manual pentesting

File: _i18n/en/_posts/blog/2024-08-21-security-tools-tech-preview.md (hunk 2 above)
Comment (90n20) [reply _i18n/en/_posts/blog/2024-08-21-security-tools-tech-preview.md]: Modified the text to clarify that we perform DAST scanning, with the option to add SAST related reports

File: _i18n/en/_posts/blog/2024-08-21-security-tools-tech-preview.md (hunk 3 above)
Comment (90n20) [reply _i18n/en/_posts/blog/2024-08-21-security-tools-tech-preview.md]: Added links to the repos

**Comment by bitkeks** [comment]
LGTM - approved from my side. Cannot approve in Github because co-author

**Review by 90n20 — COMMENTED** [review]
LGTM. 
Nice job on the post, it reflects all work done on this topic.

**Review by 90n20 — APPROVED** [review]

**Comment by bitkeks** [comment]
Waiting for review from @garloff to minimize misunderstandings in SCS security offering :slightly_smiling_face:

File: _i18n/en/_posts/blog/2024-09-03-security-tools-tech-preview.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+While this pipeline focuses primarily on dynamic testing and vulnerability management, it does not include tools specifically dedicated to Static Application Security Testing (SAST). However, some of the tools in the pipeline, such as Trivy, can be configured and customized to generate reports that resemble SAST output by analyzing container images and configurations for known vulnerabilities and security misconfigurations. This flexibility allows teams to tailor the tools to their specific security needs, though dedicated SAST tools would still be necessary for comprehensive source code analysis.
```
Comment (garloff) [inline _i18n/en/_posts/blog/2024-09-03-security-tools-tech-preview.md:30]:
```suggestion
As this pipeline focuses primarily on dynamic testing and vulnerability management, it does not include tools specifically dedicated to Static Application Security Testing (SAST). However, some of the tools in the pipeline, such as Trivy, can be configured and customized to generate reports that resemble SAST output by analyzing container images and configurations for known vulnerabilities and security misconfigurations. This flexibility allows teams to tailor the tools to their specific security needs, though dedicated SAST tools would still be beneficial for comprehensive source code analysis.
```

File: _i18n/en/_posts/blog/2024-09-03-security-tools-tech-preview.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+## Usage
+
+To utilize the automated pentesting tools in your own pipeline, you can follow the documentation provided in the [Quickstart guide](https://docs.scs.community/docs/operating-scs/components/automated-pentesting/quickstart). This guide offers detailed instructions on how to integrate the tools into your CI/CD pipeline, ensuring that your infrastructure and applications are continuously tested for vulnerabilities in line with compliance standards such as CIS Controls and NIS2.
+
+DefectDojo can be incorporated into your security operations workflow, providing a centralized platform for vulnerability management. By integrating DefectDojo, your team can more effectively track and prioritize security issues, ensuring that they are addressed in a timely manner and that the organization remains compliant with relevant cybersecurity regulations.
```
Comment (garloff) [inline _i18n/en/_posts/blog/2024-09-03-security-tools-tech-preview.md:50]:
```suggestion
DefectDojo can be incorporated into your security operations workflow, providing a centralized platform for vulnerability management. By integrating DefectDojo, your team can more effectively track and prioritize security issues, ensuring that they are addressed in a timely manner and that the organization remains secure and complies with relevant cybersecurity regulations.
```
I see security as a goal in itself and the primary motivator for addressing security issues. Compliance with regulations is another goal, but even without regulatory challenges, keeping a platform secure must be worked on with high priority.

File: _i18n/en/_posts/blog/2024-09-03-security-tools-tech-preview.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+## Implementation details
+
+The implementation of automated pentesting within SCS is designed to be both practical and effective. The code repositories are publicly available, allowing users to review and contribute to the development process. Pipelines have been implemented as a Proof of Concept (PoC) in Zuul, an open-source CI/CD system that is well-suited for complex workflows in large-scale cloud environments.
+
+One of the challenges faced during implementation was the long runtime of certain jobs, particularly those involving Greenbone Vulnerability Management. Efforts have been made to optimize these jobs to reduce their impact on overall pipeline performance. The export functionality of Trivy Operator reports has also been streamlined, enabling easier integration with other security tools and processes, thereby ensuring ongoing alignment with regulatory requirements like CIS Controls and the NIST CSF.
+
+## Usage
+
```
Comment (garloff) [inline _i18n/en/_posts/blog/2024-09-03-security-tools-tech-preview.md:47]:
```suggestion

While the SCS team runs these tools regularly against deployed SCS test environments, the tools are meant to be set up by cloud operators (CSPs) to test the security of their own test, reference and production deployments. This way, potential misconfigurations and security issues are detected before they hit production and before evil hackers find them.

```
This is stating the obvious, but better clarify another time that the SCS team does NOT pentest the CSP's SCS clouds, but does provide tooling for the operators to pentest their own.

**Review by garloff — COMMENTED** [review]
Looks good to me overall.
Good: Very clear explanation why this is useful.
Not perfect: A bit repetitive, as the statements on SAST vs DAST and the explanation of the value and limitations come three times, in the intro, in the text and in the conclusion again. Maybe intro or conclusion can be significantly shortened (or even unified and then one left out)?

I added a few very minor suggestions in addition. Feel free to merge them or disregard (if you state a reason to do so).

File: _i18n/en/_posts/blog/2024-09-03-security-tools-tech-preview.md (hunk 6 above)
Comment (garloff) [reply _i18n/en/_posts/blog/2024-09-03-security-tools-tech-preview.md:30]: To explain this:
- While expresses an *opposition*, but the fact that we don't include specific SAST tools is a *consequence* of our focus.
- Whether or not comprehensive source code analysis is necessary is in the eye of the reader. But it certainly is beneficial, so I suggest we use a bit more defensive wording here.

**Pull request merged**